### ISSUE TITLE
Foraging + Ephys metadata GUI upgrade needs

### ISSUE THREAD

**Issue by KanghoonJ** [AllenNeuralDynamics/dynamic-foraging-task#780]
I have used the dynamic foraging task GUI to generate basic ephys/behavior metadata. However, it currently doesn't cover optotagging and surface stimulation related information (light stimulation protocol, absolute and relative newscale coordinates for probe and laser stimulation targets). Currently, the most time-consuming and error-prone part for metadata generation is manual entry of all those values. 
To make metadata generation fully automated for ephys + behavior with optotagging and surface stimulation, we need a GUI development that can read absolute and relative coordinates of NewScale manipulators for probe and laser modules and write the coordinates into a json or xlsx format. 
Here are lists that I need for metadata, 
- Bregma coordinates for the probe
- Z coordinate value for surface_Z when the probe hits the surface of the brain. 
- Recording coordinates
- Bregma coordinate for the laser module
- Number of surface stimulation targets
- Newscale coordinates for each surface stimulation target 
- Light stimulation protocols for each target (power, frequency, pulse duration, # of repeats, etc.)

I'd like to request following feature developments: 
- Have a button in the GUI that selects a NewScale manipulator and automatically reads its absolute and relative coordinates. Manual entry of these values take time (difficult in the middle of experiments) and are susceptible to human error. 
- Have a field for multiple light stimulation protocol (Stimulation target, pulse duration, pulse interval, frequency, power, number of repeats, etc) 

Bugs in current metadata GUI:  
1. Putting negative values in the metadata field has a bug.
2. Loading a saved metadata file has a bug in retrieving a correct project name.

**Comment by alexpiet** [comment]
> Bugs in current metadata GUI:  
> 
> Putting negative values in the metadata field has a bug.
> Loading a saved metadata file has a bug in retrieving a correct project name.

@KanghoonJ Can you document these bugs as separate issues?

**Comment by alexpiet** [comment]
@KanghoonJ What is your timeline of when you would like this upgrade?

**Comment by alexpiet** [comment]
@JeremiahYCohen @KanghoonJ We need to discuss who should implement this feature request.

**Comment by KanghoonJ** [comment]
> > Bugs in current metadata GUI:
> > Putting negative values in the metadata field has a bug.
> > Loading a saved metadata file has a bug in retrieving a correct project name.



> > Bugs in current metadata GUI:
> > Putting negative values in the metadata field has a bug.
> > Loading a saved metadata file has a bug in retrieving a correct project name.
> 
> @KanghoonJ Can you document these bugs as separate issues?

I posted a separate issue for these bugs.

**Comment by KanghoonJ** [comment]
> @KanghoonJ What is your timeline of when you would like this upgrade?

Currently, ephys + behavior metadata is not fully integrated regarding optotagging and surface stimulation. Jeremiah wants to prioritize automation of accurate metadata generation for ephys + behavior data. I would like this can be upgraded as soon as we can (in a month?). But it needs an input from who're working on GUI upgrade. Hard to estimate time and efforts required for this upgrade.

**Comment by KanghoonJ** [comment]
> @JeremiahYCohen @KanghoonJ We need to discuss who should implement this feature request.
Sure. I will schedule a meeting for this and describe what features are needed for the upgrade.

**Comment by alexpiet** [comment]
@JeremiahYCohen @KanghoonJ and I discussed this. We decided
- This is a general need to track ephys probes, laser probes, and microscopes that is independent of this behavioral task. 
- @KanghoonJ will discuss with other ephys scientists to make a specific proposal for a tool that interacts with the newscale stages that move the ephys probes, collects probe metdata, and generates AIND metadata files. Then @KanghoonJ will ask SIPE to develop this tool
- @KanghoonJ will talk with @dyf about tools to combine metadata files from this probe tool and the behavioral task. 

Closing this issue since its not a specific request for this repository
